### PR TITLE
Added unit test to ensure watches are not re-triggered on consul reload

### PR DESCRIPTION
**Pull request by pierresouchay**
This ensures no regression about https://github.com/hashicorp/consul/issues/7318

And ensure that https://github.com/hashicorp/consul/issues/7446 cannot happen anymore

### CODE REVIEW

File: agent/agent_endpoint_test.go [hunk 1]
```diff
@@ -1342,6 +1344,126 @@ func TestAgent_Reload(t *testing.T) {
...
+// TestAgent_ReloadDoesNotTriggerWatch Ensure watches not triggered after reload
+// see https://github.com/hashicorp/consul/issues/7446
+func TestAgent_ReloadDoesNotTriggerWatch(t *testing.T) {
+	t.Parallel()
+	dc1 := "dc1"
+	checkInterval := "1s" // This value ensure we could detect as transient critical state
+	tmpFileRaw, err := ioutil.TempFile("", "rexec")
+	if err != nil {
+		t.Fatalf("failed to make tmp file: %#v", err)
+	}
```
Comment (ShimmerGlass) [inline agent/agent_endpoint_test.go]: `defer os.Remove(tmpFileRaw)`

File: agent/agent_endpoint_test.go [hunk 2]
```diff
@@ -1342,6 +1344,126 @@ func TestAgent_Reload(t *testing.T) {
...
 }
 
+// TestAgent_ReloadDoesNotTriggerWatch Ensure watches not triggered after reload
+// see https://github.com/hashicorp/consul/issues/7446
+func TestAgent_ReloadDoesNotTriggerWatch(t *testing.T) {
+	t.Parallel()
+	dc1 := "dc1"
+	checkInterval := "1s" // This value ensure we could detect as transient critical state
+	tmpFileRaw, err := ioutil.TempFile("", "rexec")
+	if err != nil {
```
Comment (ShimmerGlass) [inline agent/agent_endpoint_test.go]: opt: these can be `require.NoError(t, err)`

File: agent/agent_endpoint_test.go (hunk 1 above)
Comment (pierresouchay) [reply agent/agent_endpoint_test.go]: if does no take a file descriptor, only a path: https://golang.org/pkg/os/#Remove, so I think I am correct

File: agent/agent_endpoint_test.go (hunk 2 above)
Comment (pierresouchay) [reply agent/agent_endpoint_test.go]: DONE

**Comment by pierresouchay** [comment]
damned unstable integration tests

**Pull request merged**